### PR TITLE
Typography updates

### PR DESCRIPTION
**Pull request by vpicone**
The goal of this PR is to identify a core set of rules defining our typographic vertical rhythm on the website. It needs to scale in a way that meets the needs of 10+ sites that may or may not have the design personnel to address each page with a fine tooth comb.

Importantly, this **does not** take away the ability of individual sites to make their own spacing rules for custom components or even to override the defaults completely if their typographic needs are demonstrably different than the default accommodates

closes: https://github.com/carbon-design-system/gatsby-theme-carbon/issues/68

### CODE REVIEW

**Comment by vpicone** [comment]
I migrated the Layout page from https://www.carbondesignsystem.com/guidelines/layout

It's a solid representation of our typographic challenges at the moment. Captions are temporarily missing in from migrated version due to an oversight in moving the images over.

**Comment by vpicone** [comment]
These are some examples problem spots where content seems to be just floating in space with no real direction taken from the carbon site.

Untitled-2

Untitled-1

Untitled-3

File: packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+// Responsive by default
+.responsive {
+  // 7 columns wide
+  @include carbon--breakpoint('md') {
+    width: 62.5%;
+  }
+
+  // 5 columns wide
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss]:
```suggestion
  // 7 columns wide
```

comments are wrong, my fault 😉

File: packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@import '~@carbon/elements/scss/type/type';
+
+.paragraph {
+  @include carbon--type-style('body-long-02');
+  margin-bottom: $spacing-04;
+}
+
+// Responsive by default
+.responsive {
+  // 7 columns wide
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss]:
```suggestion
  // 5 columns wide
```

**Comment by vpicone** [comment]
@alisonjoseph yeah still a WIP, I'll ping @jeanservaas when the layout page is ready for insepection

File: packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  @include carbon--breakpoint('md') {
+    width: 62.5%;
+  }
+
+  // 5 columns wide
+  @include carbon--breakpoint('lg') {
+    width: 58.333%;
```
Comment (alisonjoseph) [inline packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss]:
```suggestion
    width: 66.6667%;
```

Comment (alisonjoseph) [reply packages/gatsby-theme-carbon/src/components/markdown/markdown.module.scss]: should be 8 per https://github.com/carbon-design-system/gatsby-theme-carbon/issues/55

**Comment by vpicone** [comment]
Going to close this and investigate alternate strategies for implementation

**Pull request closed without merging**